mlflow/models/container/__init__.py

```diff
@@ -114,6 +114,8 @@ def _install_pyfunc_deps(
             install_mlflow_cmd = ["pip install /opt/mlflow/."]
         elif install_mlflow:
             install_mlflow_cmd = [f"pip install mlflow=={MLFLOW_VERSION}"]
+        else:
+            install_mlflow_cmd = []
 
         if install_mlflow_cmd:
```

Comment:
i was also seeing that this variable is used before declaration:

https://github.com/mlflow-automation/mlflow/actions/runs/9548382619/job/26315440945#step:10:4733

Reply:
Great catch!

Reply:
It'd be nice if we could detect possibly-undefined variables like this one.

Reply:
/ai Write me a linter using the `ast` module to detect possibly-undefined variables.

```
# Example

def f():
    if foo():
        a = 1
    elif bar():
        a = 2
    
    print(a)  # -> this line throws if foo() is False and bar() is also False.
```

Reply:
It seems challenging to detect them.